prov/sockets/src/sock_ep.c

```diff
@@ -1863,13 +1868,15 @@ int sock_ep_get_conn(struct sock_ep_attr *attr, struct sock_tx_ctx *tx_ctx,
 		conn = sock_ep_connect(attr, av_index);
 
 	if (!conn) {
-		SOCK_LOG_ERROR("Error in connecting: %s\n", strerror(errno));
-		if (errno == EINPROGRESS)
+		SOCK_LOG_ERROR("Error in connecting: %s\n",
+			       strerror(ofi_sockerr()));
+		if (ofi_sockerr() == EINPROGRESS)
```

Comment:
We should push this close to the actual socket call.  This may require changing sock_ep_connect() to return an error code, rather than assuming errno is set.

Reply:
Re-worked `sock_ep_connect` function to return `INT` and instance of `struct sock_conn` as a return value. Please, review `sock_ep_connect`